### PR TITLE
Updated herald recycler view xml

### PR DESCRIPTION
**Pull request by aryanagarwal15**
This is in reference to #89 

Do inform if any more changes need to be made.

### CODE REVIEW

File: androidClient/app/src/main/java/com/csatimes/dojma/services/UpdateCheckerService.java [hunk 1]
```diff
@@ -155,10 +155,8 @@ public void execute(Realm realm) {
                                         entry.setOriginalTime(post.getString("date").substring(11));
 
                                         try {
-                                            String imageUrl = post.getString("thumbnail");
-                                            if (imageUrl == null)
-                                                imageUrl = post.getJSONObject("thumbnail_images").getJSONObject
-                                                        ("thumbnail").getString("url");
+                                            String imageUrl = post.getJSONObject("thumbnail_images").getJSONObject
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/services/UpdateCheckerService.java:158]: the update checker service is to be removed (and replaced by fcm infra)

**Comment by Rushi98** [comment]
use vector drawables for icons

**Comment by Rushi98** [comment]
The thumbnail images look distorted for the size. Maybe use `fullThumbnailImage`. See https://github.com/MobileApplicationsClub/DoJMA/blob/fe0705aa23df8a88729d8fe2b8264818910142c7/androidClient/app/src/main/java/com/csatimes/dojma/articles/ArticleViewerActivity.java#L192

**Comment by Rushi98** [comment]
There needs to be some sort of boundary for a news items. Currently the thumb image and (headline + date + icons) don't appear like part of same item.

File: androidClient/app/src/main/java/com/csatimes/dojma/herald/HeraldFragment.java [hunk 2]
```diff
@@ -38,9 +38,6 @@ public void onViewCreated(@NonNull final View view, final Bundle savedInstanceSt
         heraldRv = view.findViewById(R.id.fragment_herald_rv);
         final Context context = getContext();
         heraldRv.setLayoutManager(new GridLayoutManager(context, getGridSpan()));
-        if (context != null) {
-            heraldRv.addItemDecoration(new DividerItemDecoration(context, VERTICAL));
-        }
     }
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/herald/HeraldFragment.java:39]: remove unused imports

File: androidClient/app/src/main/java/com/csatimes/dojma/services/UpdateCheckerService.java (hunk 1 above)
Comment (aryanagarwal15) [reply androidClient/app/src/main/java/com/csatimes/dojma/services/UpdateCheckerService.java:158]: We can take care of replacing update checker service by fcm infra in another issue. This issue was meant just for changing the recycler view only. Please open a new issue for what you are suggesting here.

**Comment by aryanagarwal15** [comment]
@Rushi98 I believe the herald item is looking neater without the boundary demarcation. More or less I am following google News guidelines here. They also don't have any boundary/demarcation.

**Review by Rushi98 — COMMENTED** [review]
Rename ```article_back``` to ```thumb_placeholder```

File: androidClient/app/src/main/res/layout/item_format_herald.xml [hunk 3]
```diff
@@ -4,91 +4,76 @@
...
         app:layout_constraintTop_toBottomOf="@+id/item_format_herald_title" />
-
-    <TextView
-        android:id="@+id/item_format_herald_desc"
-        android:layout_width="0dp"
-        android:layout_height="0dp"
-        android:layout_marginEnd="8dp"
+    <View
+        android:layout_width="match_parent"
+        android:layout_height="0.5dp"
```
Comment (Rushi98) [inline androidClient/app/src/main/res/layout/item_format_herald.xml]: https://stackoverflow.com/a/21746197 suggests `1px` for these use cases.

Comment (aryanagarwal15) [reply androidClient/app/src/main/res/layout/item_format_herald.xml]: I'll make a final commit. Do u have any other suggestions. Please Acknowledge this message accordingly.

Comment (Rushi98) [reply androidClient/app/src/main/res/layout/item_format_herald.xml]: seems ok

**Comment by aryanagarwal15** [comment]
@Rushi98 All requested changes are made. Please merge with master at your convenience.

**Pull request merged**